### ISSUE TITLE
LoginWithPopup doesn't work in Microsoft Word add-in on Mac

### ISSUE THREAD

**Issue by Svn-Sp** [auth0/auth0-spa-js#1274]
### Checklist

- [X] The issue can be reproduced in the [auth0-spa-js sample app](https://github.com/auth0-samples/auth0-javascript-samples/tree/master/01-Login) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-spa-js#readme), [Examples](https://github.com/auth0/auth0-spa-js/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-spa-js/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://auth0.com/docs/libraries/auth0-single-page-app-sdk) and [API documentation](https://auth0.github.io/auth0-spa-js/), and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-spa-js/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

We are developing a Microsoft Word add-in and would like to use auth0 to authenticate our users.

**Our steps so far**:
1. First we tried to use `LoginWithRedirect`. However, since the add-in is running in an Iframe inside of MS Word it doesn't work and we cannot switch to the classical authentication experience (where we could theoretically enable displaying the dialog in an iframe) since we need to support enterprise connections.
2. After that we tried to use `LoginWithPopup`. This works fine for MS Word in Web on Linux and Windows and for MS Word Desktop on Windows. It _doesn't_ open a popup for MS Word Desktop on Mac though. It's a [known issue](https://stackoverflow.com/questions/48091227/get-different-behavior-using-window-open-and-window-loaction-in-word-add-i)  that `window.open()` will behave unreliably for MS Word add-ins and the proposed solution for it is to use the Office [Dialog Api](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/dialog-api-in-office-add-ins). (But this is not supported by the auth0 package)

How can we still integrate Auth0 in our product?

**tl;dr**
We have an app running in an iframe and need to support enterprise connections. Due to restrictions of the environment the app runs in, the default `LoginWithPopup` will fail because we can't rely on the `window.open()` method

### Reproduction

1. Create Javascript app for Microsoft Office add-in
2. Add button calling auth0 `LoginWithPopup` Method
3. Run app on Microsoft Word Desktop on Mac
-> The popup won't open

### Additional context

**Old sample project**
There is [this](https://github.com/OfficeDev/Office-Add-in-Auth0) sample project from 2019 but it uses the less-secure method of retrieving the access token directly and it supports _only_ third party providers.

**Other,  unanswered post about the same issue**
https://community.auth0.com/t/how-to-intregrate-auth0-in-office-add-in-with-react/102392

### auth0-spa-js version

2.1.3

### Which framework are you using (React, Angular, Vue...)?

React (but should be irrelevant)

### Framework version

18.3.1 (but should be irrelevant)

### Which browsers have you tested in?

The issue occurs when running on MS Word Desktop on Mac, which uses _Safari Webkit_. For Firefox on Linux and Edge on Windows it works.

**Comment by frederikprijck** [comment]
Closing, feel free to let us know if we can help further.

**Comment by kcarnold** [comment]
I didn't see this issue earlier, but my team ended up implementing basically the same solution: https://github.com/AIToolsLab/writing-tools/blob/ef3f7400596bca6c7a5142847e1bf414f0180cc8/frontend/src/api/wordEditorAPI.ts#L4

The solution is much more awkward than necessary, I wrote https://github.com/auth0/auth0-spa-js/issues/1321 about it.

(Is this repo even maintained anymore?)
